### PR TITLE
feat: add TreeConfig

### LINKED ISSUE #9821: Make tree settings configurable
currently we rely on constants for certain tree settings

https://github.com/paradigmxyz/reth/blob/c4bf5bb8ca271a6055f5af8431fd0bfb4ba79feb/crates/engine/tree/src/tree.rs#L52-L58

it would be useful to have a way to configure these easily, we already have a way to configure certain values:

https://github.com/paradigmxyz/reth/blob/c4bf5bb8ca271a6055f5af8431fd0bfb4ba79feb/crates/engine/tree/src/tree.rs#L261-L262
https://github.com/paradigmxyz/reth/blob/c4bf5bb8ca271a6055f5af8431fd0bfb4ba79feb/crates/engine/tree/src/tree.rs#L445-L447

## TODO
* introduce a treeconfig type that has these as fields, and defaults are the constants

### PR DESCRIPTION
**Pull request by fgimenez**
Closes #9821

The configuration is done at the launcher level (which adds a new dep to `reth-ethereum-node`), we could do it at the service level too, let me know wdyt

### CODE REVIEW

File: crates/engine/tree/src/tree/config.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+}
+
+/// Engine tree configuration builder.
+#[derive(Debug, Default)]
+pub struct TreeConfigBuilder {
+    persistence_threshold: Option<u64>,
+    block_buffer_limit: Option<u32>,
+    max_invalid_header_cache_length: Option<u32>,
+}
```
Comment (mattsse) [inline crates/engine/tree/src/tree/config.rs:62]: I think the builder type is redundant here if we move the with setters to Treeconfig directly so you can configure it like `TreeConfig::default().with_().with()`

**Review by mattsse — CHANGES_REQUESTED** [review]
one suggestion

and we can re-export that type from etherem-engine like we did with other types already

https://github.com/paradigmxyz/reth/blob/cf7698ab035bcb8209ea6cd422feace7ce37fc47/crates/ethereum/engine/src/service.rs#L13-L16

File: crates/engine/tree/src/tree/config.rs (hunk 1 above)
Comment (fgimenez) [reply crates/engine/tree/src/tree/config.rs]: makes sense, done

**Review by mattsse — APPROVED** [review]
very nice

**Pull request merged**